validator_worker/src/follower.rs

```diff
@@ -24,24 +24,20 @@ enum NewStateResult {
 pub async fn tick<A: Adapter + 'static>(iface: &SentryApi<A>) -> Result<(), Box<dyn Error>> {
     let from = iface.channel.spec.validators.leader().id.clone();
     let new_msg_response = iface.get_latest_msg(from, "NewState".to_string()).await?;
-    let new_msg = new_msg_response
-        .msg
-        .get(0)
-        .and_then(|message_types| match message_types {
-            MessageTypes::NewState(new_state) => Some(new_state.clone()),
-            _ => None,
-        });
+    let new_msg = match new_msg_response {
+        Some(MessageTypes::NewState(new_state)) => Some(new_state),
+        _ => None,
+    };
+
     let our_latest_msg_response = iface
         .get_our_latest_msg("ApproveState+RejectState".to_string())
```

Comment:
perhaps this api would be nicer if it gets `&[&str]` so that we avoid the `+` hack